pkg/cloud/amazon/common.go

```diff
@@ -65,14 +66,57 @@ func ResolveRegionWithoutOptions() (string, error) {
 	return ResolveRegion("", "")
 }
 
+// GetClusterNameFromAWS uses the AWS SDK to parse through each EKS cluster until it finds one that matches the endpoint in
+// the kubeconfig. From there it will retrieve the cluster name
+func GetClusterNameFromAWS(profileOption string, regionOption string, kubeEndpoint string) (string, error) {
```

Comment:
👍 That's pretty much what I had in mind :-)

@dgozalo, WDYT? To me, that seems more reliable.